pvlib/data/gold/sdm.py

```diff
@@ -0,0 +1,847 @@
+    device.a_ref = device.a_ref / Ns_module
+    device.R_s = device.R_s / Ns_module
+    device.R_sh_ref = Ns_module * device.R_sh_ref
+
+    return device
+
+
+def make_gold_dataset():
+    """
+    Make a gold dataset of I-V curves that accurately solve single-diode model.
+
+    The set of I-V curves are generated from a range of PV devices: small c-Si
+    cell, large c-Si cell, c-Si "mini-module", c-Si module, and CdTe module.
+    Additionally, each device is ideal, nominal, or degraded in terms of its
+    series and parallel resistances. I-V curves for each device are generated
+    over a matrix of irradiance and temperature. The I-V curve data are
+    returned in a python dict that is in a lossless, machine-interoperable
+    format, which is the expected input to most of the other functions in this
+    python module. (Other functions can convert this dict to a human-readable
+    format.) An explict method of Bishop [1] is used to compute (V, I) pairs
+    mostly in the first quadrant, and interval analysis guarantees the
+    numerical computations up to a specified tolerance.
+
+    New gold data can be added to the returned dict. Each change should be
+    versioned in ``dataset_version``, following SemVer and maintaining
+    backwards compatibility whenever possible. Be sure to update all dependent
+    functions.
+
+    Parameters
+    ----------
+
+    Returns
+    -------
+    gold_dataset : dict
+        The gold data are returned in a dict with metadata (e.g., timestamp,
+        versions, units) and a list of representative PV devices, where each
+        device is a dict that includes a list of dict containing the diode
+        equation coefficients and corresponding gold I-V curve data.
+
+    References
+    ----------
+    [1] J. W. Bishop, "Computer simulation of the effects of electrical
+    mismatches in photovoltaic cell interconnection circuits", Solar Cells,
+    vol 25, pp. 73-89, 1988.
+    """
+
+    # Load module inside function to avoid a global dependency requirement
+    # http://pyinterval.readthedocs.io/en/latest/guide.html
+    # Use `pip install pyinterval`, even in conda distro's
+    from interval import interval, imath
+
+    # Approximate largest possible absolute residual value, determined
+    #  empirically
+    tol = 1.e-12
+
+    tol_interval = interval([-tol, tol])
+
+    # Generate I-V curves, each with same number of points, over matrix of
+    #  (G,T) combinations
+    G_list = [100., 200., 400., 600., 800., 1000., 1100.]  # W/m^2
+    T_list = [15., 25., 50., 75.]  # degC
+    NUM_PTS = 21
+
+    # The "base" devices, having ideal, normal, and degraded versions
+    devices = [
+        bootstrap_si_device_bishop(),
+        bootstrap_si_device_bishop(N_s=8),
+        bootstrap_si_cell_large(),
+        pvlib.pvsystem.retrieve_sam('cecmod').SunPower_SPR_E20_327,
+        pvlib.pvsystem.retrieve_sam('cecmod').First_Solar_FS_495
+    ]
+
```

Comment:
What do you think about copying the parameters for all 5 devices into golden_devices.csv or something similar?  Then you just need a single read.